lib/board.rb

```diff
@@ -0,0 +1,49 @@
+class Board
+  attr_accessor :cells
+
+  def initialize
+    reset!
+  end
+
+  def reset!
+    @cells = Array.new(9, " ")
+  end
+
+  def display
+    puts " #{@cells[0]} | #{@cells[1]} | #{@cells[2]} "
+    puts "-----------"
+    puts " #{@cells[3]} | #{@cells[4]} | #{@cells[5]} "
+    puts "-----------"
+    puts " #{@cells[6]} | #{@cells[7]} | #{@cells[8]} "
+  end
+
+  def position(input)
+    @cells[input.to_i - 1]
+  end
+
+  def taken?(input)
+    @cells[input.to_i - 1] != " "
+  end
+
+  def valid_move?(input)
+    if input.to_i.between?(1,9) && !taken?(input)
+      return true
+    else
+      puts "I'm afraid I can't let you do that."
+      return false
+    end
+  end
+
+  def full?
+    !@cells.include?(" ")
+  end
+
+  def turn_count
+    9 - @cells.count(" ")
```

Comment:
Interesting solution!  You an also use `cells.count { |cell| cell == "X" || cell == "O" }`